HLTrigger/HLTfilters/plugins/HLTTripletMass.cc

```diff
@@ -0,0 +1,224 @@
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "HLTTripletMass.h"
+
+#include "DataFormats/Common/interface/Handle.h"
+
+#include "DataFormats/Common/interface/Ref.h"
+#include "DataFormats/HLTReco/interface/TriggerFilterObjectWithRefs.h"
+
+//#include "DataFormats/Candidate/interface/Particle.h"
+#include "DataFormats/RecoCandidate/interface/RecoEcalCandidate.h"
+#include "DataFormats/RecoCandidate/interface/RecoEcalCandidateFwd.h"
+#include "DataFormats/EgammaCandidates/interface/Electron.h"
+#include "DataFormats/EgammaCandidates/interface/ElectronFwd.h"
+#include "DataFormats/RecoCandidate/interface/RecoChargedCandidate.h"
+#include "DataFormats/RecoCandidate/interface/RecoChargedCandidateFwd.h"
+#include "DataFormats/L1TCorrelator/interface/TkMuon.h"
+#include "DataFormats/L1TCorrelator/interface/TkMuonFwd.h"
+#include "DataFormats/L1TParticleFlow/interface/PFTau.h"
+#include "DataFormats/L1TParticleFlow/interface/HPSPFTau.h"
+#include "DataFormats/L1TParticleFlow/interface/HPSPFTauFwd.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "DataFormats/Math/interface/deltaR.h"
+#include "DataFormats/Candidate/interface/Particle.h"
+#include <cmath>
+
+//
+// constructors and destructor
+//
+template <typename T1, typename T2, typename T3>
+HLTTripletMass<T1, T2, T3>::HLTTripletMass(const edm::ParameterSet& iConfig)
+    : HLTFilter(iConfig),
+      originTag1_(iConfig.template getParameter<std::vector<edm::InputTag>>("originTag1")),
+      originTag2_(iConfig.template getParameter<std::vector<edm::InputTag>>("originTag2")),
+      originTag3_(iConfig.template getParameter<std::vector<edm::InputTag>>("originTag3")),
+      inputTag1_(iConfig.template getParameter<edm::InputTag>("inputTag1")),
+      inputTag2_(iConfig.template getParameter<edm::InputTag>("inputTag2")),
+      inputTag3_(iConfig.template getParameter<edm::InputTag>("inputTag3")),
+      inputToken1_(consumes<trigger::TriggerFilterObjectWithRefs>(inputTag1_)),
+      inputToken2_(consumes<trigger::TriggerFilterObjectWithRefs>(inputTag2_)),
+      inputToken3_(consumes<trigger::TriggerFilterObjectWithRefs>(inputTag3_)),
```

Comment:
```suggestion
      inputToken1_(consumes<trigger::TriggerFilterObjectWithRefs>(iConfig.template getParameter<edm::InputTag>("inputTag1"))),
      inputToken2_(consumes<trigger::TriggerFilterObjectWithRefs>(iConfig.template getParameter<edm::InputTag>("inputTag2"))),
      inputToken3_(consumes<trigger::TriggerFilterObjectWithRefs>(iConfig.template getParameter<edm::InputTag>("inputTag3"))),
```
(There is no need to persist the inputTag's, which are not used anywhere else in the code)